### PR TITLE
Module quagga command

### PR DESCRIPTION
**Pull request by le9i0nx**
##### ISSUE TYPE

 - New Module Pull Request

##### ANSIBLE VERSION

```
ansible 2.2.1.0
  config file = /home/le9i0nx/crypt/ansible-home/ansible.cfg
  configured module search path = ['./library']

```

##### ADDITIONAL INFORMATION

```
Adds a module that works with the vtysh command line

vtysh is integrated shell of Quagga software.

http://www.nongnu.org/quagga/docs/quagga.html#VTY-CLI-Commands
```

### CODE REVIEW

**Comment by gundalow** [comment]
Hi,
Can you please move this to `net_tools, ie `lib/ansible/modules/net_tools/quagga/`

File: lib/ansible/modules/network/quagga/quagga_command.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: quagga_command
+version_added: 2.4
+short_description: Provides configuring quaggas
+description:
+    - The M(quagga_command) module takes the command name followed by a list of space-delimited arguments.
```
Comment (gundalow) [inline lib/ansible/modules/network/quagga/quagga_command.py]: `Takes the command name followed by a list of space-delimited arguments.`

No need to repeat the module name

File: lib/ansible/modules/network/quagga/quagga_command.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    - The given command will be executed on all selected nodes.
+    - Documentation for M (quagga) can be found U(http://www.nongnu.org/quagga/docs.html)
```
Comment (gundalow) [inline lib/ansible/modules/network/quagga/quagga_command.py]: `M(...)` will generate a link to a module page, which in this case doesn't exist, so I think this should be

`Documentation for C(quagga) can be found U(http://www.nongnu.org/quagga/docs.html)`

http://docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html#formatting-options

File: lib/ansible/modules/network/quagga/quagga_command.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+options:
+    raw:
+        description:
+            - vtysh command after 'configure terminal'
```
Comment (gundalow) [inline lib/ansible/modules/network/quagga/quagga_command.py]: Perhaps
`vtysh command after C(configure terminal).`

**Comment by le9i0nx** [comment]
Hi,
Thanks for the specified defects in the documentation.

**Comment by maxamillion** [comment]
This seems like a better candidate for a role than a module to me, I don't understand what the module provides over a small task list in a role that you pass variables to.

**Comment by le9i0nx** [comment]
@maxamillion  Sorry, I did not understand the offer. it is too complicated.
@gundalow can we already move this further?

**Comment by maxamillion** [comment]
I disagree that this is too complicated to make into a role, @gundalow thoughts?

**Comment by gundalow** [comment]
Looks like there are a number of existing Quagga roles in Galaxy https://galaxy.ansible.com/list#/roles?page=1&page_size=10&autocomplete=quagga&order=-stargazers_count,name

**Comment by le9i0nx** [comment]
Well, honestly I watched these roles. And I did not like the fact that they require a complete description of the configuration.
And this role allows using partial config via clt.

**Comment by LensHunnel** [comment]
Hi,

This module will be integrate to ansible ? the fact that this module allow to add a partial config via clt is just an advantage that the others roles doesn't have.

**Comment by Akasurde** [comment]
Please submit new module requests to https://github.com/ansible-collections/community.general. Thanks for your contribution.

**Pull request closed without merging**